### DISCUSSION TITLE
Is changing the shell from bash not recommended?

### DISCUSSION THREAD

**Discussion by ghost** [msys2/MSYS2-packages#4510] [General]
I want to change the shell from `bash` to `mksh`.

**Comment by sskras** [comment]
I think it just works:

![image](https://github.com/msys2/MSYS2-packages/assets/7887758/ef7a18a2-7522-4b76-af4f-9ba3f717f93e)

I used following command:
```
$ sed -i '/^'$USER'/s/bash/mksh/' /etc/passwd
```

Then maybe this looks more readable:
```
$ sed -i "/^${USER}/s/bash/mksh/" /etc/passwd
```

**Reply by sskras** [reply]
@iahung2 wrote:

> `sed: can't read /etc/passwd: No such file or directory`

Please excuse me – I forgot that the file needs to be generated at first:
https://sourceforge.net/p/msys2/discussion/general/thread/76612760/#0392
```sh
/usr/bin/mkpasswd  >  /etc/passwd
```
